holidays/countries/fiji.py

```diff
@@ -0,0 +1,128 @@
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import SAT, SUN, OCT, NOV
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+    TUE_TO_PREV_MON,
+)
+
+
+class Fiji(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """
+    References:
+    - https://www.fiji.gov.fj/About-Fiji/Public-Holidays
+    - https://www.timeanddate.com/holidays/fiji/
+    - https://en.wikipedia.org/wiki/List_of_festivals_in_Fiji
+    """
+
+    country = "FJ"
+    weekend = {SAT, SUN}
+    supported_categories = (PUBLIC, WORKDAY)
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    start_year = 1970
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
+        kwargs.setdefault(
+            "observed_rule", SUN_TO_NEXT_MON + SAT_SUN_TO_NEXT_MON_TUE + TUE_TO_PREV_MON
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        dts_observed = set()
+
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday("Good Friday")
+
+        # Easter Saturday.
+        self._add_holy_saturday("Easter Saturday")
+
+        # Easter Sunday.
+        self._add_easter_sunday("Easter Sunday")
+
+        # Easter Monday.
+        self._add_easter_monday("Easter Monday")
+
+        # National Youth Day
+        self._add_holiday_may_4("National Youth Day")
+
+        if self._year >= 2023:
+            # Girmit Day (starting 2023).
+            self._add_observed(self._add_holiday_may_14("Girmit Day"))
+
+        # Fiji Day.
+        self._add_holiday_oct_10("Fiji Day")
+
+        # Prophet Mohammed's Birthday
+        dts_observed.update(self._add_mawlid_day("Prophet Mohammed's Birthday"))
+
+        # https://www.timeanddate.com/holidays/fiji/diwali
+        diwali_dates = {
+            2015: (NOV, 11),
+            2016: (OCT, 27),
+            2017: (OCT, 16),
+            2018: (NOV, 7),
+            2019: (OCT, 28),
+            2020: (NOV, 14),  # also should be a day off on Nov 16
+            2021: (NOV, 4),
+            2022: (OCT, 25),
+            2023: (NOV, 13),
+            2024: (NOV, 1),
+            2025: (OCT, 21),
+        }
+
+        # Diwali
+        if self._year in diwali_dates:
+            self._add_holiday("Diwali", diwali_dates[self._year])
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day("Christmas Day"))
+
+        # Boxing Day.
+        self._add_observed(self._add_christmas_day_two("Boxing Day"))
```

Comment:
This eliminates weird cases like `"1972-12-25": "Boxing Day (observed); Christmas Day"` and `"1979-12-24": "Christmas Day (observed)"` that shouldn't have existed.
```suggestion
        # Christmas Day.
        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)

        # Boxing Day.
        self._add_observed(self._add_christmas_day_two("Boxing Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
```